app/lib/features/space/sheets/link_space_sheet.dart

```diff
@@ -0,0 +1,228 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/widgets/default_button.dart';
+import 'package:acter/common/widgets/search.dart';
+import 'package:acter/common/widgets/side_sheet.dart';
+import 'package:acter/common/widgets/spaces/select_space_form_field.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class LinkSpacePage extends ConsumerStatefulWidget {
+  final String? parentSpaceId;
+
+  const LinkSpacePage({super.key, required this.parentSpaceId});
+
+  @override
+  ConsumerState<LinkSpacePage> createState() => _LinkSpacePageConsumerState();
+}
+
+class _LinkSpacePageConsumerState extends ConsumerState<LinkSpacePage> {
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+  final List<String> subSpacesIds = [];
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      selectParentSpaceData();
+    });
+    super.initState();
+  }
+
+  //Select parent space data
+  void selectParentSpaceData() {
+    if (widget.parentSpaceId != null) {
+      final notifier = ref.read(selectedSpaceIdProvider.notifier);
+      notifier.state = widget.parentSpaceId;
+    }
+  }
+
+  //Fetch known sub-spaces list of selected parent space
+  void fetchKnownSubspacesData() {
+    final selectedParentSpaceId = ref.watch(selectedSpaceIdProvider);
+    if (selectedParentSpaceId == null) return;
+    final sp = ref.watch(spaceRelationsOverviewProvider(selectedParentSpaceId));
+    sp.when(
+      data: (space) {
+        subSpacesIds.clear();
+        for (int i = 0; i < space.knownSubspaces.length; i++) {
+          subSpacesIds.add(space.knownSubspaces[i].getRoomId().toString());
+        }
+      },
+      error: (e, s) => Container(),
+      loading: () => Container(),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
+    //Fetch known sub-spaces list of selected parent space
+    fetchKnownSubspacesData();
```

Comment:
fetchKnownSubspacesData() is for fetching selected parentSpace's child list. 

We have provide option to change the selected parentSpace in this screen. So in that case we need to fetch newly selected parentSpace's child list every time selection change. So that's why I have putted it here.

Please help me with your thoughts here in this scenario.

Reply:
why not just have it triggered and updated based on user action? like ```onPressed(), onTap()``` which is on selection...

Reply:
Let me check what can be done here. Thanks for sharing thoughts :)